src/renderer/src/components/song/song-item/SongItem.tsx

```diff
@@ -57,32 +51,59 @@ const SongItem: Component<SongItemProps> = ({
   });
 
   return (
-    <div
-      class="group relative isolate select-none rounded-md"
-      classList={{
-        "outline outline-2 outline-accent": selectedSong().path === song.path,
-      }}
-      data-active={selectedSong().path === song.path}
-      ref={item}
-      data-url={song.bg}
-      onContextMenu={showMenu}
+    <Popover
+      isOpen={localShow}
+      onValueChange={setLocalShow}
+      middlewares={[flip(), offset({ crossAxis: 30 })]}
+      placement="right"
+      offset={15}
     >
-      <SongImage
-        class="absolute inset-0 z-[-1] h-full w-full rounded-md bg-cover bg-center bg-no-repeat opacity-30 group-hover:opacity-90"
+      <Portal>
+        <Popover.Overlay />
+        <Popover.Content
+          onClick={(e) => {
+            e.stopImmediatePropagation();
+            setLocalShow(false);
+          }}
+        >
+          {...children}
```

Comment:
I don't think this the dropdown content should be retrieved from the children, could you move it to another prop instead, something like:

``` tsx
<SongItem
ContextMenu={<SongContextMenu>
                  <Show when={isQueueExist() === true}>
                    <PlayNext path={s.path} />
                  </Show>
                  <AddToPlaylist path={s.path} />
                </SongContextMenu>}
 />
```